kclvm/parser/src/entry.rs

```diff
@@ -0,0 +1,413 @@
+///
+/// # Error
+/// The package root path for package name `__main__` is only used once. If there are multiple
+/// package root paths for `__main__`, an error `conflict kcl.mod file` is returned.
+///
+/// # Example
+///
+/// ```rust
+/// use std::path::PathBuf;
+/// use kclvm_config::modfile::get_compile_entries_from_paths;
+/// let testpath = PathBuf::from("./src/testdata/multimods").canonicalize().unwrap();
+///
+/// // [`kcl1_path`] is a normal path of the package [`kcl1`] root directory.
+/// // It looks like `/xxx/xxx/xxx`.
+/// let kcl1_path = testpath.join("kcl1");
+///
+/// // [`kcl2_path`] is a mod relative path of the packege [`kcl2`] root directory.
+/// // It looks like `${kcl2:KCL_MOD}/xxx/xxx`
+/// let kcl2_path = PathBuf::from("${kcl2:KCL_MOD}/main.k");
+///
+/// // [`kcl3_path`] is a mod relative path of the [`__main__`] packege.
+/// let kcl3_path = PathBuf::from("${KCL_MOD}/main.k");
+///
+/// // [`external_pkgs`] is a map to show the real path of the mod relative path [`kcl2`].
+/// let mut external_pkgs = std::collections::HashMap::<String, String>::new();
+/// external_pkgs.insert("kcl2".to_string(), testpath.join("kcl2").to_str().unwrap().to_string());
+///
+/// // [`get_compile_entries_from_paths`] will return the map of package name to package root real path.
+/// let entries = get_compile_entries_from_paths(
+///     &[
+///         kcl1_path.to_str().unwrap().to_string(),
+///         kcl2_path.display().to_string(),
+///         kcl3_path.display().to_string(),
+///     ],
+/// external_pkgs).unwrap();
+///
+/// assert_eq!(entries.len(), 3);
+///
+/// assert_eq!(entries.get_nth_entry(0).unwrap().name(), "__main__");
+/// assert_eq!(
+///     PathBuf::from(entries.get_nth_entry(0).unwrap().path())
+///         .canonicalize()
+///         .unwrap()
+///         .display()
+///         .to_string(),
+///     kcl1_path.canonicalize().unwrap().to_str().unwrap()
+/// );
+///
+/// assert_eq!(entries.get_nth_entry(1).unwrap().name(), "kcl2");
+/// assert_eq!(
+///     PathBuf::from(entries.get_nth_entry(1).unwrap().path())
+///         .canonicalize()
+///         .unwrap()
+///         .display()
+///         .to_string(),
+///     testpath
+///         .join("kcl2")
+///         .canonicalize()
+///         .unwrap()
+///         .to_str()
+///         .unwrap()
+/// );
+///
+/// assert_eq!(entries.get_nth_entry(2).unwrap().name(), "__main__");
+/// assert_eq!(
+///     PathBuf::from(entries.get_nth_entry(2).unwrap().path())
+///         .display()
+///         .to_string(),
+///     kcl1_path.join("main.k").canonicalize().unwrap().to_str().unwrap()
+/// );
+///
+/// ```
+pub fn get_compile_entries_from_paths(
+    file_paths: &[String],
+    opts: &LoadProgramOptions,
+) -> Result<Entries, String> {
+    if file_paths.is_empty() {
+        return Err("No input KCL files or paths".to_string());
+    }
+    let mut result = Entries::default();
+    let mut m = std::collections::HashMap::<String, String>::new();
```

Comment:
Is there a problem not using `IndexMap` here?  What does `m` mean?

Reply:
removed it.